src/services/car/__tests__/leasing.test.ts

```diff
@@ -22,4 +22,117 @@ describe("CAR service", () => {
+      beforeEach(() => {
+        fetchMock.mockResponse("")
+      })
+
+      describe("validate only", () => {
+        it("calls validation endpoint", async () => {
+          await sendLeasingInterest(12345, leasingInterest(), {
+            validateOnly: true,
+          })
+
+          expect(fetch).toHaveBeenCalledWith(
+            expect.stringMatching(
+              /\/listings\/12345\/leasing\/interests\/validate$/
+            ),
+            expect.any(Object)
+          )
+        })
+
+        it("returns a success", async () => {
+          const result = await sendLeasingInterest(12345, leasingInterest(), {
+            validateOnly: true,
+          })
+
+          expect(result).toEqual({ tag: "success", result: leasingInterest() })
+        })
+      })
+
+      describe("submit", () => {
+        it("calls submission endpoint", async () => {
+          await sendLeasingInterest(12345, leasingInterest())
+
+          expect(fetch).toHaveBeenCalledWith(
+            expect.stringMatching(/\/listings\/12345\/leasing\/interests$/),
+            expect.any(Object)
+          )
+        })
+
+        it("sends recaptcha token in a header", async () => {
+          await sendLeasingInterest(12345, leasingInterest(), {
+            recaptchaToken: "token",
+          })
+
+          expect(fetch).toHaveBeenCalledWith(expect.any(String), {
+            headers: expect.objectContaining({
+              "Recaptcha-Token": "token",
+            }),
+            body: expect.any(String),
+            method: "POST",
+          })
+        })
+
+        it("returns a success", async () => {
+          const result = await sendLeasingInterest(12345, leasingInterest())
+
+          expect(result).toEqual({ tag: "success", result: leasingInterest() })
+        })
+      })
+    })
+
+    describe("invalid parameters", () => {
+      const errors = [{ param: "email", error: "validations.invalid-format" }]
+
+      beforeEach(() => {
+        fetchMock.mockResponse(
+          JSON.stringify({ errors, message: "validations.not-valid" }),
+          {
+            status: 400,
+          }
+        )
+      })
+
+      it("returns a failure", async () => {
+        const result = await sendLeasingInterest(
+          12345,
+          leasingInterest({ email: "test@test" })
+        )
+
+        expect(result.tag).toEqual("error")
+        if (result.tag === "error") {
+          expect(result.errors).toEqual(errors)
+          expect(result.message).toEqual("validations.not-valid")
+        }
+      })
+    })
+
+    describe("other error", () => {
+      beforeEach(() => {
+        fetchMock.mockResponse("", {
+          status: 500,
+        })
+      })
+
+      it("returns a failure", async () => {
+        const result = await sendLeasingInterest(12345, leasingInterest())
+
+        expect(result.tag).toEqual("error")
+        if (result.tag === "error") {
+          expect(result.message).toEqual("validation.other-error")
```

Comment:
Shouldn't this be covered by the unit test of the `handleValidationError` function?

Reply:
Oh, I get what you mean above now. I guess we could test that we call `handleValidationError` instead, but I was never a big fan of testing whether the error handling method was called, but would rather test the result. WDYT?

Reply:
Yeah it's fine just for my understanding, no strong preference :)